scipy/spatial/transform/rotation.py

```diff
@@ -866,3 +868,106 @@ def __getitem__(self, indexer):
+            arbitrary rotation to `vectors`. Shape must match `vectors`.
+        weights : None or array_like, 1D
+            Coefficients describing the relative importance of the vectors in
+            `vectors`. Number of values in `weights` must match number of
+            vectors specified in `vectors` and `transformed_vectors`. If None,
+            then all values in `weights` are assumed equal to +1.
+            Default is None.
+
+        Returns
+        -------
+        estimated_rotation : `Rotation` instance
+            Contains a single rotation which is the best estimate of the
+            rotation that transforms `vectors` to `transformed_vectors`.
+        sensitivity_matrix : `numpy.ndarray`, shape (3, 3)
+            Scaled covariance matrix of the three component error vector
+            between `transformed_vectors` and the result of applying
+            `estimated_rotation` on `vectors`.
+
+        References
+        ----------
+        .. [1] F. Landis Markley,
+                “Attitude determination using vector observations: a fast
+                optimal matrix algorithm,” Journal of Astronautical Sciences,
+                Vol. 41, No.2, 1993, pp. 261-280.
+        """
+        vectors = np.asarray(vectors)
+        if vectors.ndim != 2 or vectors.shape[-1] != 3:
+            raise ValueError("Expected input `vectors` to have shape (N, 3)"
+                             ", got {}.".format(vectors.shape))
+        if vectors.shape[0] == 1:
+            raise ValueError("Rotation cannot be estimated using a single "
+                             "vector.")
+
+        outvecs = np.asarray(transformed_vectors)
+        if outvecs.ndim != 2 or outvecs.shape[-1] != 3:
+            raise ValueError("Expected input `transformed_vectors` to have "
+                             "shape (N, 3), got {}".format(outvecs.shape))
+
+        if vectors.shape != outvecs.shape:
+            raise ValueError("Expected inputs `vectors` and "
+                             "`transformed_vectors` to have same shapes, got "
+                             "{} and {} respectively.".format(
+                                vectors.shape, outvecs.shape))
+
+        if weights is None:
+            weights = np.ones(vectors.shape[0])
+        else:
+            weights = np.asarray(weights)
+            if weights.ndim != 1:
+                raise ValueError("Expected `weights` to be 1 dimensional, got "
+                                 "shape {}.".format(weights.shape))
+            if weights.shape[0] != vectors.shape[0]:
+                raise ValueError("Expected `weights` to have number of values "
+                                 "equal to number of input vectors, got "
+                                 "{} values and {} vectors.".format(
+                                    weights.shape[0], vectors.shape[0]))
+
+        B = np.einsum('ji,jk->ik', weights[:, None] * outvecs, vectors)
```

Comment:
I think this is a slight overuse of `einsum`. There are `np.outer` and other stuff, which might be more efficient. I don't know, perhaps nothing is wrong in using `einsum`, but I somehow don't like it. Judge yourself.

Reply:
I looked at `np.outer` and that performs outer product of 1D vectors. When given an array of vectors like in our application, it performs outer products of every possible pair.
We need to perform outer products of corresponding vectors and sum the resultant matrices. I think this is a succinct way to do it. I may be wrong here; I'll change it if I find a better way.